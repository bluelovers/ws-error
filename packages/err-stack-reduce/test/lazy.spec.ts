import errStackReduceCore from '../index';
import { basename } from 'path';

describe(basename(__filename), () =>
{

	test(`test`, () =>
	{
		let mainError = new Error();
		let error = new Error();

		let { fn, actual } = _check(mainError, error);

	});

})

function _check(mainError: Error, error: Error)
{
	let fn = errStackReduceCore(mainError);
	let actual = fn(error);

	expect(actual.prefix).toContain(error.name);
	expect(actual.stack.length).toBeGreaterThan(0);
	expect(actual.stack).toMatch(/^\s+at /);

	expect(actual).toMatchSnapshot({
		stack: expect.any(String),
		originalStack: expect.any(String),
	});

	return {
		fn,
		actual,
	}
}
