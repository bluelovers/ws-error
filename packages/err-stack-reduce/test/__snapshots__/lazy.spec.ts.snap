// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy.spec.ts test 1`] = `
Object {
  "error": [Error],
  "message": "",
  "originalStack": Any<String>,
  "prefix": "Error: ",
  "rawTrace": Any<Array>,
  "stack": Any<String>,
  "type": "Error",
}
`;
